_posts/en/newsletters/2020-09-30-newsletter.md

```diff
@@ -0,0 +1,133 @@
+---
+title: 'Bitcoin Optech Newsletter #117'
+permalink: /en/newsletters/2020/09/30/
+name: 2020-09-30-newsletter
+slug: 2020-09-30-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a compiler bug that casts doubt on the
+safety of secure systems and includes our regular sections with popular
+questions and answers from the Bitcoin StackExchange, announcements of
+releases and release candidates, and summaries of notable changes to
+popular Bitcoin infrastructure software.
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Discussion about compiler bugs:** a [test][oconnor test] written
+  last week by Russell O'Connor for libsecp256k1 failed due to a
+  [bug][gcc bug] in the GNU C Compiler's (GCC's) built-in version of the
+  standard `memcmp` (memory compare) function.  This function takes two
+  regions of memory, interprets them as integer values, and returns
+  whether the first region is less than, equal to, or greater than the
+  second region.  This is a commonly used low-level function.
+  Programs are almost never designed to verify such fundamental
+  operations were performed correctly, so bugs of this type can easily
+  result in a program producing incorrect results.  Those incorrect
+  results are possible even if the program's code was well reviewed,
+  strongly tested, formally verified, and otherwise built with the
+  utmost care.  A single incorrect result in the execution of
+  cryptographic or consensus code could result in serious consequences
+  for the users of that code---or anyone who depends on remaining in
+  consensus with users of that code.
+
+    The potential issues affect not only software written in C and
+    compiled with GCC, but any software or library that depends on
+    software or libraries compiled with the affected versions of GCC.
+    That may be a significant amount of software on affected systems,
+    such as recent versions of [Ubuntu and Fedora][maxwell systems].  It
+    includes programs written in other languages whose compilers or
+    interpreters were built using GCC.
+
+    Issues were opened in both the [libsecp256k1][libsecp256k1 #823] and
+    [Bitcoin Core][bitcoin core #20005] repositories to find and
+    mitigate any effects of this bug.  The topic was also
+    [discussed][irc memcmp] during the weekly Bitcoin Core Developers
+    Meeting.  More broadly, Pieter Wuille [asked][wuille tweet] for
+    suggestions about a "process that can give assurances for correct
```

Comment:
I interpreted this tweet as being rhetorical, i.e. "if the tools we're depending on to build our applications are so fundamentally broken, how can we have any assurance at all that the software is correct?"